### DISCUSSION TITLE
Can neper plot the crystal orientation distribution?

### DISCUSSION THREAD

**Discussion by fzw9381** [neperfepx/neper#256] [Q&A] [answered]
Can neper plot the crystal orientation distribution? 
I saw the distribution map of the crystal orientation on the official website. Is that one generated by neper?

**Comment by CsatiZoltan** [comment]
Yes, it can. If the orientations are given in your mesh file within the
```
$ElsetOrientations
...
$EndElsetOrientations
```
section, you can colour the element sets based on the crystal orientations by using
```
neper -V your_mesh.msh -dataelset3dcol ori -print output
```
Check Section 5.1.5 in the [manual](https://neper.info/docs/neper.pdf).

**Comment by mkasemer** [comment]
Perhaps @fzw9381 is referring to "orientation distributions", for example on pole figures (fiber distributions), rather than spatial distributions of orientations plotted on the FE mesh?

**Comment by yguilhem** [comment]
That might have been generated using [mtex](https://mtex-toolbox.github.io/) (matlab toolbox) or something similar.

**Comment by rquey** [comment]
Pole figures (and other orientation figures) are plotted using an in-house code. That in-house code generates top-quality figures in vectorial format (with Latex fonts). Something I can push to Neper in the mid term if there is interest.

**Reply by abarun22** [reply]
Quite interested in plotting pole figures if included in Neper. I can see i have such requirements in my upcoming work. Any references to that in-house code most welcome.

**Comment by mkasemer** [comment, marked as answer]
All, (and most recently @abarun22). Neper indeed now has capabilities to plot orientation distribution functions in Rodrigues space (and perhaps Euler-Bunge space? @rquey would have to confirm), on pole figures, and inverse pole figures for a number of crystal symmetries. The tutorials give a lot of details and should get you set on the right track. Coupled with FEPX for deformation simulations, this helps close a huge gap in the "holistic" workflow. See the open-access article [https://iopscience.iop.org/article/10.1088/1757-899X/1249/1/012021/pdf](https://iopscience.iop.org/article/10.1088/1757-899X/1249/1/012021/pdf) for a description of the Neper/FEPX workflow, which affords near-complete automation "top to bottom".

@abarun22 : Regarding your specific question, please look at [https://neper.info/doc/tutorials/pole_figure.html](https://neper.info/doc/tutorials/pole_figure.html) for more information.

**Reply by abarun22** [reply]
Thanks Matthew. Appreciated.
